squirrel/orders/views.py

```diff
@@ -63,30 +63,108 @@ def get_table_data(self):
             return Team.objects.filter(members=self.request.user)
 
 
+# Not a View.
+def _state_helper(request, order_object):
+    if order_object.state == "REQ":
+        if request.user.has_perm("orders.change_order") or request.user.has_perm(
+            "orders.approve_order"
+        ):
+            my_states = [("REQ", "Requested"), ("APP", "Approved")]
+        else:
+            my_states = [("REQ", "Requested")]
+    elif order_object.state == "APP":
+        if request.user.has_perm("orders.change_order") or request.user.has_perm(
+            "orders.receive_order"
+        ):
+            my_states = [("APP", "Approved"), ("REA", "Ready for pick-up")]
+        else:
+            my_states = [("APP", "Approved")]
+    elif order_object.state == "REA":
+        if request.user.has_perm("orders.change_order") or request.user.has_perm(
+            "orders.complete_order"
+        ):
+            my_states = [("REA", "Ready for pick-up"), ("COM", "Completed")]
+        else:
+            my_states = [("REA", "Ready for pick-up")]
+    else:  # order_object.state is COM
+        my_states = [("COM", "Completed")]
+    return my_states
+
+
 @login_required
 def order(request, order_id=None):
     if order_id:
         order_object = get_object_or_404(Order, id=order_id)
     else:
         order_object = None
 
+    # limit team field
     if request.user.has_perm("orders.view_team"):
         my_teams = Team.objects.all()
     else:
         my_teams = Team.objects.filter(members=request.user)
 
     if request.method == "POST":
         if order_object:
-            if (
-                request.user.has_perm("orders.change_order")
-                or request.user in order_object.team.members.all()
+            if request.user.has_perm("orders.change_order"):
+                # can do almost anything
+                form = OrderForm(
+                    request.POST,
+                    instance=order_object,
+                    teams=my_teams,
+                    states=Order.STATE_CHOICES,
+                )
+            elif (
```

Comment:
The body of almost all `elif` statements here is the same, they should be combined into one elif statement with `or`

Reply:
done in ec7e3f567ab185294d81bd0d8bc0616d7896ded7